pkg/handler/v2/operatingsystemprofile/operatingsystemprofile.go

```diff
@@ -0,0 +1,171 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operatingsystemprofile
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/go-kit/kit/endpoint"
+
+	apiv1 "k8c.io/kubermatic/v2/pkg/api/v1"
+	apiv2 "k8c.io/kubermatic/v2/pkg/api/v2"
+	"k8c.io/kubermatic/v2/pkg/handler/middleware"
+	"k8c.io/kubermatic/v2/pkg/handler/v1/common"
+	"k8c.io/kubermatic/v2/pkg/provider"
+	osmv1alpha1 "k8c.io/operating-system-manager/pkg/crd/osm/v1alpha1"
+)
+
+// TODO: Find a way to populate these dynamically.
+// These OSPs are created after cluster creation by OSM. We need a way to display these
+// before cluster creation. For example, these are required in the cluster creation wizard for
+// KKP dashboard.
+// Namespace is purposefully left empty since we cannot determine namespace of these resources before cluster creation.
+var defaultOperatingSystemProfiles = []apiv2.OperatingSystemProfile{
+	{
+		Name: "osp-amzn2",
+	},
+	{
+		Name: "osp-centos",
+	},
+	{
+		Name: "osp-flatcar",
+	},
+	{
+		Name: "osp-rhel",
+	},
+	{
+		Name: "osp-rockylinux",
+	},
+	{
+		Name: "osp-sles",
+	},
+	{
+		Name: "osp-ubuntu",
+	},
+}
+
+// listOperatingSystemProfilesReq defines HTTP request for listOperatingSystemProfilesForCluster
+// swagger:parameters listOperatingSystemProfilesForCluster
+type listOperatingSystemProfilesReq struct {
+	common.ProjectReq
+	// in: path
+	ClusterID string `json:"cluster_id"`
+}
+
+// GetSeedCluster returns the SeedCluster object.
+func (req listOperatingSystemProfilesReq) GetSeedCluster() apiv1.SeedCluster {
+	return apiv1.SeedCluster{
+		ClusterID: req.ClusterID,
+	}
+}
+
+func DecodeListOperatingSystemProfiles(c context.Context, r *http.Request) (interface{}, error) {
+	var req listOperatingSystemProfilesReq
+
+	clusterID, err := common.DecodeClusterID(c, r)
+	if err != nil {
+		return nil, err
+	}
+	req.ClusterID = clusterID
+
+	projectReq, err := common.DecodeProjectRequest(c, r)
+	if err != nil {
+		return nil, err
+	}
+	req.ProjectReq = projectReq.(common.ProjectReq)
+
+	return req, nil
+}
+
+func ListOperatingSystemProfilesEndpointForCluster(userInfoGetter provider.UserInfoGetter, operatingSystemProfile provider.OperatingSystemProfileProvider) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (response interface{}, err error) {
+		req := request.(listOperatingSystemProfilesReq)
+
+		userClusterNamespace, err := clusterNamespaceFromContext(ctx, userInfoGetter, req.ProjectID, req.ClusterID)
+		if err != nil {
+			return nil, err
+		}
+
+		ospList, err := operatingSystemProfile.ListUnsecured(ctx, userClusterNamespace)
+		if err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
```

Comment:
This wont work when the clusters are on Seeds which are not also the Master cluster. You need a dynamic client